pkg/mocks/plugin.go

```diff
@@ -3,33 +3,101 @@ package mocks
 import (
 	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 const (
 	Namespace         = "default"
-	StableServiceName = "argo-rollouts-stable-service"
-	CanaryServiceName = "argo-rollouts-canary-service"
-	HTTPProxyName     = "argo-rollouts-httpproxy"
+	StableServiceName = "argo-rollouts-stable"
+	CanaryServiceName = "argo-rollouts-canary"
+
+	HTTPProxyName               = "argo-rollouts"
+	ValidHTTPProxyName          = "argo-rollouts-valid"
+	OutdatedHTTPProxy           = "argo-rollouts-outdated"
+	InvalidHTTPProxyName        = "argo-rollouts-invalid"
+	FalseConditionHTTPProxyName = "argo-rollouts-false-condition"
+
+	HTTPProxyGeneration    = 1
+	HTTPProxyDesiredWeight = 20
 )
 
-var HTTPProxyObj = contourv1.HTTPProxy{
-	ObjectMeta: metav1.ObjectMeta{
-		Name:      HTTPProxyName,
-		Namespace: Namespace,
-	},
-	Spec: contourv1.HTTPProxySpec{
-		Routes: []contourv1.Route{
+func MakeObjects() []runtime.Object {
+	httpProxy := newValidHttpProxy(HTTPProxyName)
+
+	validHttpProxy := newValidHttpProxy(ValidHTTPProxyName)
+
+	invalidHttpProxy := newValidHttpProxy(InvalidHTTPProxyName)
+	invalidHttpProxy.Status = contourv1.HTTPProxyStatus{
+		Conditions: []contourv1.DetailedCondition{
+			{
+				Condition: contourv1.Condition{
+					Type:               contourv1.ConditionTypeServiceError,
+					Status:             contourv1.ConditionTrue,
+					ObservedGeneration: HTTPProxyGeneration,
+				},
```

Comment:
be a helper function under `pkg/utils`?

Reply:
As of right now, I think it's best to keep it simple since there are only two cases in the same function that need to build the `Conditions` slice. If there are future changes that require the same behaviour, we can do the refactor then.